src/swap/utils/processSwapTransaction.ts

```diff
@@ -41,6 +50,39 @@ export async function processSwapTransaction({
       confirmations: 1,
     });
     setPendingTransaction(false);
+
+    // for the V2 API, we use Uniswap's `UniversalRouter`, which uses `Permit2` for ERC-20 approvals
+    // this adds an additional transaction/step to the swap process
+    // since we need to make an extra transaction to `Permit2` to allow the UniversalRouter to spend the approved funds
+    // this would typically be a (gasless) signature, but we're using a transaction here to allow batching for Smart Wallets
+    // read more: https://blog.uniswap.org/permit2-and-universal-router
+    if (!useAggregator) {
+      setPendingTransaction(true);
+      const permit2ContractAbi = parseAbi([
+        'function approve(address token, address spender, uint160 amount, uint48 expiration) external',
+      ]);
+      const data = encodeFunctionData({
+        abi: permit2ContractAbi,
+        functionName: 'approve',
+        args: [
+          quote.from.address as Address,
+          UNIVERSALROUTER_CONTRACT_ADDRESS,
+          BigInt(quote.fromAmount),
+          20_000_000_000_000,
```

Comment:
Why this number? what's the meaning behind.

Reply:
https://docs.uniswap.org/contracts/permit2/reference/allowance-transfer

it's the deadline where the approval is no longer valid - since we're only approving the exact amount `fromAmount` it shouldn't be a security concern if this isn't granular

Reply:
I feel like we should either have a comment here, or have a variable that better explain what's for.

This will help future developer undersand what's going on.

Reply:
added a comment with a link to the above